File: LeetCode/49字母异位词分组.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> u_map;
        for (auto &x : strs) {
            string str = x;
            sort(str.begin(), str.end());
            u_map[str].emplace_back(x);
        }
        vector<vector<string>> ans;
        for (auto iter = u_map.begin(); iter != u_map.end(); iter++) {
            ans.emplace_back(iter->second);
        }
        return ans;
    }
};

File: LeetCode/49字母异位词分组.go
// sort.Slice对切片进行排序，需要重写compare方法
func groupAnagrams(strs []string) [][]string {
	hash := map[string][]string{}
	for _, v := range strs {
		s := []byte(v)
		sort.Slice(s, func(i, j int) bool {
			return s[i] < s[j]
		})
		str := string(s)
		hash[str] = append(hash[str], v)
	}
	ans := make([][]string, 0, len(hash))
	for _, v := range hash {
		ans = append(ans, v)
	}
	return ans
}